holoviews/operation/element.py

```diff
@@ -316,8 +316,12 @@ def _process(self, matrix, key=None):
         dx = np.diff(data, 1, axis=1)[0:r-1, 0:c-1]
         dy = np.diff(data, 1, axis=0)[0:r-1, 0:c-1]
 
-        cyclic_range = 1.0 if not matrix_dim.cyclic else matrix_dim.range
+        cyclic_range = None if not matrix_dim.cyclic else np.diff(matrix_dim.range)
```

Comment:
Using ``np.diff(matrix_dim.range)`` if fine assuming that both values are numeric. This assumption may be violated as the range tuple is allowed to use a ``None`` value to indicate lower/upper bounds that are not set. For this reason, I would have an extra line with something like:

```if  matrix_dim.cyclic and (None in matrix_dim.range):
   raise Exception('Cyclic range must be specified to compute the gradient over cyclic quantities')